### ISSUE TITLE
nvalid template Arguments for std::make_optional in Eclipse 2024-03 with GCC version 13.2.0 (C++23)

### ISSUE THREAD

**Issue by VietHungPhan** [eclipse-cdt/cdt#741]
Hi,
My OS is Red Hat Enterprise Linux 8
I always get the error "Cannot instantiate template function" for std::make_optional in Eclipse 2024-03 with linux GCC version 13.2.0 (C++23). Below is my codes.

![image](https://github.com/eclipse-cdt/cdt/assets/43219867/2bf49de8-7c06-4706-9294-83cd984e8bf3)

Thanks.

**Comment by i-garrison** [comment]
Yes I can reproduce this now with some `GCC 13.2.1` and `-std=c++17`.

**Comment by VietHungPhan** [comment]
If it works fine for me with gcc/11.3.0 -std=c++23. Should I close the ticket? Thanks.
![image](https://github.com/eclipse-cdt/cdt/assets/43219867/d57fd0ff-d27b-4e65-8818-94ebeb315f92)
BR,
/Hung
